torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1250,11 +1250,14 @@ struct to_ir {
     return graph->create(kind, n_outputs)->setSourceRange(loc);
   }
 
-  Value* emitTernaryIf(const TernaryIf& expr) {
+  Value* emitTernaryIf(
+      const TernaryIf& expr,
+      const TypePtr& type_hint = nullptr) {
     CondValue cond_value = emitCondExpr(expr.cond());
     auto true_expr = [&] { return emitExpr(expr.true_expr()); };
```

Comment:
Can we add the type hint into the `emitExpr` of `true_expr` and `false_expr` ? I think that would make it so we don't need to touch `emitListComprehension` 

Can you add a test like `Optional[int] = None if cond else 1` to show that logic works not just for `Any` ?

Reply:
Added type hint into `emitExpr`, and moved `unCheckedCast` to it.